python-checks/src/test/java/org/sonar/python/checks/ExpressionsTest.java

```diff
@@ -83,6 +88,85 @@ public void remove_parentheses() {
+    assertThat(unescape(stringElement("'a\\\nb'"))).isEqualTo("ab");
+    // ignored windows line break
+    assertThat(unescape(stringElement("'a\\\r\nb'"))).isEqualTo("ab");
+    // ignored mac line break
+    assertThat(unescape(stringElement("'a\\\r'"))).isEqualTo("a");
+    assertThat(unescape(stringElement("'a\\\rb'"))).isEqualTo("ab");
+
+    assertThat(unescape(stringElement("'\\\\'"))).isEqualTo("\\");
+    assertThat(unescape(stringElement("'0\\\\'"))).isEqualTo("0\\");
+    assertThat(unescape(stringElement("'\\\\0'"))).isEqualTo("\\0");
+
+    assertThat(unescape(stringElement("'\\'\\\"\\''"))).isEqualTo("'\"'");
+    assertThat(unescape(stringElement("'\\a'"))).isEqualTo("\u0007");
+    assertThat(unescape(stringElement("'\\b'"))).isEqualTo("\u0008");
+    assertThat(unescape(stringElement("'\\f'"))).isEqualTo("\u000C");
+    assertThat(unescape(stringElement("'\\n'"))).isEqualTo("\n");
+    assertThat(unescape(stringElement("'\\r'"))).isEqualTo("\r");
+    assertThat(unescape(stringElement("'\\t'"))).isEqualTo("\t");
+    assertThat(unescape(stringElement("'\\v'"))).isEqualTo("\u000B");
+
+    assertThat(unescape(stringElement("'\\x48\\u0065\\U0000006c\\x6c\\x6f\\x20\\x57\\x6f\\x72\\x6c\\x64\\x21'")))
+      .isEqualTo("Hello World!");
+
+    assertThat(unescape(stringElement("b'\\x48\\u0065\\U0000006c\\x6c'"))).isEqualTo("H\\u0065\\U0000006cl");
+
+    assertThat(unescape(stringElement("'\\0'"))).isEqualTo("\u0000");
+    assertThat(unescape(stringElement("'\\1'"))).isEqualTo("\u0001");
+    assertThat(unescape(stringElement("'\\7'"))).isEqualTo("\u0007");
+    assertThat(unescape(stringElement("'\\10'"))).isEqualTo("\u0008");
+    assertThat(unescape(stringElement("'a\\777b'"))).isEqualTo("a\u01FFb");
+    assertThat(unescape(stringElement("'a\\7777b'"))).isEqualTo("a\u01FF7b");
+    assertThat(unescape(stringElement("'a\\78b'"))).isEqualTo("a\u00078b");
+    assertThat(unescape(stringElement("'\\o12'"))).isEqualTo("\n");
+    assertThat(unescape(stringElement("'\\o5 \\o6'"))).isEqualTo("\u0005 \u0006");
+
+    // unsupported escape
+    assertThat(unescape(stringElement("'\\N{LATIN CAPITAL LETTER A}'"))).isEqualTo("\\N{LATIN CAPITAL LETTER A}");
+
+    // invalid escape
+    assertThat(unescapeString("\\", true)).isEqualTo("\\");
+    assertThat(unescape(stringElement("'\\o'"))).isEqualTo("\\o");
+    assertThat(unescape(stringElement("'\\z'"))).isEqualTo("\\z");
+    assertThat(unescape(stringElement("'\\X00'"))).isEqualTo("\\X00");
+    assertThat(unescape(stringElement("'\\x0'"))).isEqualTo("\\x0");
+    assertThat(unescape(stringElement("'\\xZZ\\x04'"))).isEqualTo("\\xZZ\u0004");
+    assertThat(unescape(stringElement("'\\u000'"))).isEqualTo("\\u000");
+    assertThat(unescape(stringElement("'\\U0000000'"))).isEqualTo("\\U0000000");
+
+    // Python error: f-string expression part cannot include a backslash
+    assertThat(unescape(stringElement("f'name:\\n{na\\\nme}'"))).isEqualTo("name:\n{name}");
```

Comment:
I don't see a test for the various prefixes, e.g. `r'...'`, `rb'...'`, `b'...'`...